modules/ts/include/opencv2/ts.hpp

```diff
@@ -588,3 +588,127 @@ int main(int argc, char **argv) \
 #endif
 
 #include "opencv2/ts/ts_perf.hpp"
+
+#ifdef WINRT
+#ifndef __FSTREAM_EMULATED__
+#define __FSTREAM_EMULATED__
+#include <stdlib.h>
+#include <fstream>
+#include <sstream>
+
+#undef ifstream
+#undef ofstream
+#define ifstream ifstream_emulated
+#define ofstream ofstream_emulated
+
+namespace std {
+
+class ifstream : public stringstream
+{
+    FILE* f;
+public:
+    ifstream(const char* filename)
+    : f(NULL)
+    {
+        printf("Open file (read): %s\n", filename);
+        f = fopen(filename, "r");
+        if (f == NULL)
+        {
+            printf("Can't open file: %s\n", filename);
+            return;
+        }
+        fseek(f, 0, SEEK_END);
+        size_t sz = ftell(f);
+        if (sz > 0)
+        {
+            char* buf = (char*)malloc(sz);
+            fseek(f, 0, SEEK_SET);
+            if( fread(buf, 1, sz, f) == sz )
+            {
+                this->str(std::string(buf, sz));
+            }
+            free(buf);
+        }
+    }
+
+    ifstream(const char* filename, ios_base::openmode mode)
```

Comment:
Please use here: `mode = ios_base::in`. This will allow to remove the first constructor with duplicated code.